### PR TITLE
Minor tweaks for internal rearchitecture

### PR DESCRIPTION
**Pull request by npawelek**
These changes provide continued functionality of maas with the recent
consumption and architectural decisions around RPCO deployments.

Signed-off-by: Nathan Pawelek <nathan.pawelek@rackspace.com>

### CODE REVIEW

**Comment by npawelek** [comment]
recheck_trusty_newton_deploy

**Comment by npawelek** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by npawelek** [comment]
recheck

**Comment by npawelek** [comment]
recheck_bionic_mnaio_osp13_deploy
recheck_bionic_rocky_deploy
recheck_trusty_liberty_deploy
recheck_trusty_newton_deploy
recheck_xenial_queens_deploy
recheck_xenial_rocky_deploy

File: playbooks/templates/rax-maas/cinder_volume_check.yaml.j2 [hunk 1]
```diff
@@ -2,6 +2,9 @@
 {% if (ansible_local['maas']['general']['deploy_osp'] | bool) | default(False) %}
 {% set cinder_volume_host = item.value.host %}
 {% set backend_name = cinder_volume_host.split('@')[1] %}
+{% elif (rbd_cinder_backend | default(False) | bool) and (item.key == 'ceph' or item.value.volume_backend_name == 'ceph') %}
```
Comment (BjoernT) [inline playbooks/templates/rax-maas/cinder_volume_check.yaml.j2]: I don't  think we need `and (item.key == 'ceph' or item.value.volume_backend_name == 'ceph')` as the volume backend can be different, especially with multiple ceph pools and OSA just hard codes it once it sees the rbd driver enabled

**Comment by npawelek** [comment]
recheck_trusty_newton_deploy

**Comment by npawelek** [comment]
recheck_xenial_ceph_deploy

**Comment by npawelek** [comment]
recheck_xenial_ceph_deploy

**Comment by npawelek** [comment]
recheck_xenial_ceph_deploy

**Pull request merged**